localstack/services/cloudformation/engine/template_deployer.py

```diff
@@ -361,7 +366,7 @@ def _resolve_refs_recursively(
             none_values = [v for v in join_values if v is None]
             if none_values:
                 raise Exception(
-                    "Cannot resolve CF fn::Join %s due to null values: %s" % (value, join_values)
+                    f"Cannot resolve CF Fn::Join {value} due to null values: {join_values}"
```

Comment:
Here we're using the full canonical version while using !Ref above. Would suggest keeping this consistent. (Personal preference would be slightly towards canonical)